File: examples/python/core/callbacks/notify-success.py
import boto3
import json

lambda_client = boto3.client('lambda')

# When external system succeeds
callback_id = "abc123-callback-id-from-durable-function"
result_data = json.dumps({'status': 'approved', 'amount': 1000}).encode('utf-8')

lambda_client.send_durable_execution_callback_success(
    CallbackId=callback_id,
    Result=result_data
)

File: examples/java/core/callbacks/notify-success.java
import com.fasterxml.jackson.databind.ObjectMapper;
import software.amazon.awssdk.core.SdkBytes;
import software.amazon.awssdk.services.lambda.LambdaClient;
import software.amazon.awssdk.services.lambda.model.SendDurableExecutionCallbackSuccessRequest;

LambdaClient lambdaClient = LambdaClient.create();
ObjectMapper mapper = new ObjectMapper();

// When external system succeeds
String callbackId = "abc123-callback-id-from-durable-function";
byte[] resultData = mapper.writeValueAsBytes(Map.of("status", "approved", "amount", 1000));

lambdaClient.sendDurableExecutionCallbackSuccess(
    SendDurableExecutionCallbackSuccessRequest.builder()
        .callbackId(callbackId)
        .result(SdkBytes.fromByteArray(resultData))
        .build());
